### PR TITLE
default to table separator, update mock data names

### PR DESCRIPTION
**Pull request by joe-s-avaya**
[Link to updated Table stories](https://deploy-preview-535--neo-react-library-storybook.netlify.app/?path=/story/components-table--editable-data)

## Summary by CodeRabbit

- **New Features**
	- Row separators are now displayed by default in the Table component, enhancing visual clarity.
  
- **Improvements**
	- Updated mock data entries for improved clarity and consistency, making names more straightforward.

- **Tests**
	- Adjusted test assertions to reflect changes in row height and search functionality, ensuring accurate behavior of the Table component.

### CODE REVIEW

File: src/components/Table/Table.test.jsx [hunk 1]
```diff
@@ -73,19 +73,19 @@ describe("Table", () => {
 		it("modifies the row height when the select is changed", async () => {
 			render(<Table rowHeight="compact" {...FilledFields} />);
 
-			expect(screen.getByRole("table").classList).toHaveLength(2);
+			expect(screen.getByRole("table").classList).toHaveLength(3);
```
Comment (joe-s-avaya) [inline src/components/Table/Table.test.jsx:76]: we default to having the line separator, thus we have an extra class

File: src/components/Table/Table.test.jsx [hunk 2]
```diff
@@ -689,13 +689,13 @@ describe("Table", () => {
 				FilledFields.translations.toolbar.searchInputPlaceholder,
 			);
 			await user.click(searchInput);
-			await user.keyboard("a");
+			await user.keyboard("d");
```
Comment (joe-s-avaya) [inline src/components/Table/Table.test.jsx:692]: the names changed and thus I had to update the tests to account for that

**Comment by matthewAvaya** [comment]
Rows in hover, selected, and selected-hover states should have a white horizontal line at the bottom as shown in the example below. This [guideline](https://design.avaya.com/components/table#table-states) can also be found on the Neo portal site. 
Screen Shot 2024-09-16 at 9 51 09 AM

**Review by yangAtSpoken — APPROVED** [review]

**Comment by joe-s-avaya** [comment]
> Rows in hover, selected, and selected-hover states should have a white horizontal line at the bottom as shown in the example below. This [guideline](https://design.avaya.com/components/table#table-states) can also be found on the Neo portal site.

@matthewAvaya, [Jira Issue created](https://avaya.atlassian.net/browse/NEO-2389) and pushed to the top of the backlog.

**Comment by matthewAvaya** [comment]
> > Rows in hover, selected, and selected-hover states should have a white horizontal line at the bottom as shown in the example below. This [guideline](https://design.avaya.com/components/table#table-states) can also be found on the Neo portal site.
> 
> @matthewAvaya, [Jira Issue created](https://avaya.atlassian.net/browse/NEO-2389) and pushed to the top of the backlog.

Sounds good. I will approve.

**Pull request merged**